agents.md

```diff
@@ -0,0 +1,171 @@
+- Implement parameter processing and validation
+- Handle both query and mutation execution paths
+- Transform Neptune Analytics results to NetworkX-compatible format
+
+#### Step 3: Add Query Builder Functions
+Add query building functions to `nx_neptune/clients/opencypher_builder.py`:
+- `{algorithm_name}_query()` - Build openCypher query for read operations
+- `{algorithm_name}_mutation_query()` - Build mutation query for write operations
+- Handle parameter mapping between NetworkX and Neptune Analytics
+- Return tuple of (query_string, parameter_map)
+
+#### Step 4: Add Constants
+Add algorithm-specific constants to `nx_neptune/clients/neptune_constants.py`:
+- Algorithm mutation constant (`{ALGORITHM_NAME}_MUTATE_ALG`)
+- Response field constants (`RESPONSE_{FIELD_NAME}`)
+- Parameter constants (`PARAM_{PARAMETER_NAME}`)
+
+#### Step 5: Update Module Exports
+Add algorithm to `nx_neptune/algorithms/{category}/__init__.py`:
+- Import the new algorithm function
+- Add to `__all__` list for proper module exports
+
+#### Step 6: Create Comprehensive Tests
+Create `tests/algorithms/{category}/test_{algorithm_name}.py`:
+- Test class with descriptive name
+- Mock NeptuneGraph fixture with sample return data
+- Test basic algorithm execution
+- Test with various parameter combinations
+- Test mutation operations (write_property)
+- Test error handling and edge cases
+- Verify NetworkX compatibility
+
+#### Step 7: Add Documentation
+Update `algorithms.md` with algorithm documentation:
+- Algorithm description and purpose
+- Link to Neptune Analytics documentation
+- Source file location
+- Parameter descriptions with types and defaults
+- Return value format and structure
+- Usage examples
+
+#### Step 8: Integration Testing
+- Test with actual Neptune Analytics instance using environment variables
+- Verify NetworkX compatibility and result format consistency
+- Performance benchmarking against NetworkX native implementation
+- Edge case handling (empty graphs, disconnected components)
+- Parameter validation and error scenarios
+
+#### Step 9: Update Package Exports
+Add to main `nx_neptune/__init__.py`:
+- Import the algorithm from its module
+- Add to main package `__all__` list
+- Ensure proper backend registration
+
+#### Key Implementation Guidelines
+- **Parameter Mapping**: Map NetworkX parameters to Neptune Analytics equivalents
+- **Result Transformation**: Convert Neptune Analytics JSON results to NetworkX format
+- **Error Handling**: Handle AWS service exceptions and parameter validation
+- **Logging**: Use structured logging with algorithm name for debugging
+- **Documentation**: Include Neptune Analytics documentation links and examples
+- **Testing**: Cover both query and mutation variants with comprehensive test cases
+- **Performance**: Consider result size and memory usage for large graphs
+- **Compatibility**: Ensure results match NetworkX behavior and format expectations
+
+### Instance Management
```

Comment:
Instance management is defined by the NeptuneConfig located in the nx-plugin/config.py file. New instance management steps should be included and documented in the NeptuneConfig class.

Reply:
Instance Management task execution is defined in the nx_neptune/instance_management.py file. 
Control of the task is defined in nx_neptune/utils/decorators.py